src/sentry/rules/actions/base.py

```diff
@@ -60,6 +63,37 @@ def after(self, event: Event, state: EventState) -> Generator[CallbackFuture, No
         pass
 
 
+class SentryAppEventAction(EventAction, abc.ABC):
+    """Abstract class to ensure that actions in SENTRY_APP_ACTIONS have all required methods"""
+
+    @abc.abstractmethod
+    def get_custom_actions(self, project: Project) -> Sequence[Mapping[str, Any]]:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def self_validate(self) -> None:
+        raise NotImplementedError()
+
+
+def trigger_alert_rule_action_creators(actions: Sequence[Mapping[str, str]]) -> str | None:
```

Comment:
This function will be used by `project_rule_details.py` and `project_rules.py` since it doesn't relate to the rest of the endpoint code.

Reply:
We're trying to manage the way we fire the AlertRuleActionCreators and AlertRuleActionRequesters since they make 3rd-party API calls. Having a dedicated method will work best for this.